map/framework.cpp

```diff
@@ -1132,12 +1132,13 @@ bool Framework::QueryMayBeSkipped(search::SearchParams const & params,
 
 void Framework::LoadSearchResultMetadata(search::Result & res) const
 {
-  if (res.m_metadata.m_isInitialized)
+  if (res.m_metadata.m_isInitialized || res.GetResultType() != search::Result::RESULT_FEATURE)
```

Comment:
On the other hand, we can treat non feature results as having empty metadata, and in this case we can (as you mentioned) reduce number of redundant calls. In any case, if this code will gone away, both approaches seems good to me now.